deployment/data-streams/changeset/sequence/deploy_data_streams.go

```diff
@@ -0,0 +1,312 @@
+package sequence
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/mcms"
+
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/utils/mcmsutil"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/changeset"
+	feemanager "github.com/smartcontractkit/chainlink/deployment/data-streams/changeset/fee-manager"
+	rewardmanager "github.com/smartcontractkit/chainlink/deployment/data-streams/changeset/reward-manager"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/changeset/types"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/changeset/verification"
+	dsutil "github.com/smartcontractkit/chainlink/deployment/data-streams/utils"
+)
+
+// DeployDataStreamsDestinationChainChangeset deploys the entire data streams destination chain contracts. It should be kept up to date
+// with the latest contract versions and deployment logic.
+var DeployDataStreamsDestinationChainChangeset = deployment.CreateChangeSet(deployDataStreamsLogic, deployDataStreamsPrecondition)
+
+type DeployDataStreamsConfig struct {
+	ChainsToDeploy map[uint64]DeployDataStreams
+}
+
+type BillingConfig struct {
+	LinkTokenAddress   common.Address
+	NativeTokenAddress common.Address
+	Surcharge          uint64
+}
+
+type BillingFeature struct {
+	Enabled bool
+	Config  *BillingConfig
+}
+
+type DeployDataStreams struct {
+	VerifierConfig verification.SetConfig
+
+	Billing   BillingFeature
+	Ownership types.OwnershipFeature
+}
+
+func deployDataStreamsLogic(e deployment.Environment, cc DeployDataStreamsConfig) (deployment.ChangesetOutput, error) {
+	newAddresses := deployment.NewMemoryAddressBook() // changeset output expects only new addresses
+	// Clone env. to avoid mutation the changeset Applier expects no changes.
+	existingAddresses, err := e.ExistingAddresses.Addresses()
+	abClone := deployment.NewMemoryAddressBookFromMap(existingAddresses)
+	cloneEnv := deployment.Environment{
+		Name:              e.Name,
+		Logger:            e.Logger,
+		ExistingAddresses: abClone,
+		Chains:            e.Chains,
+		SolChains:         e.SolChains,
+		NodeIDs:           e.NodeIDs,
+		Offchain:          e.Offchain,
+		OCRSecrets:        e.OCRSecrets,
+		GetContext:        e.GetContext,
+	}
+
+	var timelockProposals []mcms.TimelockProposal
+
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to get existing addresses: %w", err)
+	}
+	for chain, cfg := range cc.ChainsToDeploy {
```

Comment:
nit: This is fantastic and looks like it's doing everything we need, step by step. Great work!
It's possible to make it a little easier on the reader by breaking parts of it out into functions, so they are a little more compartmentalised and digestible when reading.

Reply:
refactored. Although not implemented here I am leaning in the direction of moving towards having some internal separate between evm/solana/etc.. other chain families. But we don't need to do that now

Reply:
I agree with that - we'll probably need it at some point, as they might need very different handling.
I also agree that we don't need it right this moment.

Reply:
I just reviewed the refactored version - amazing work! It is vastly easier to read and navigate now. Thanks!